BUILD.gn

```diff
@@ -48,7 +48,7 @@ group("browser_dependencies") {
 group("brave") {
   deps = []
   if (is_android) {
-    deps += [ "//chrome/android:chrome_public_apk" ]
+    deps += [ "//brave/build/android:brave_apk" ]
```

Comment:
I think we should move `"build/$target_os:brave",` above into `deps = []` and change the target name from `brave_apk` to `brave` so we can keep them all more standardized. wdyt?

Reply:
so we'd just have `if (!is_android) { deps += [ :generate...`

Reply:
that was resolved and re reviewed